WebDriverAgentLib/Routing/FBResponsePayload.m

```diff
@@ -103,35 +103,39 @@
 
 inline NSDictionary *FBDictionaryResponseWithElement(XCUIElement *element, BOOL compact)
 {
-  id<FBXCElementSnapshot> snapshot = element.lastSnapshot ?: element.fb_cachedSnapshot ?: [element fb_takeSnapshot:YES];
-  NSDictionary *compactResult = FBToElementDict((NSString *)[FBXCElementSnapshotWrapper wdUIDWithSnapshot:snapshot]);
-  if (compact) {
-    return compactResult;
-  }
+  __block NSDictionary *finalResult = nil;
+  @autoreleasepool {
+    id<FBXCElementSnapshot> snapshot = element.lastSnapshot ?: element.fb_cachedSnapshot ?: [element fb_takeSnapshot:YES];
+    NSDictionary *compactResult = FBToElementDict((NSString *)[FBXCElementSnapshotWrapper wdUIDWithSnapshot:snapshot]);
+    if (compact) {
+      return finalResult = compactResult;
```

Comment:
this syntax is confusing. Consider splitting variable assigning and the return expression